### PR TITLE
opensync: Support Mesh configuration

### PR DESCRIPTION
**Pull request by yashvardhan-netex**
- Patch to support basic batman configuraion through ovsdb interface
   in addition to the defaults.

Signed-off-by: Yashvardhan <yashvardhan@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 1]
```diff
@@ -980,119 +1006,172 @@ void vif_hs20_update(struct schema_Hotspot20_Config *hs2conf)
...
+				blobmsg_add_u32(b1, "mtu", 1532);
+			else
+				blobmsg_add_u32(b1, "mtu", atoi(val));
+		}
+	}
+}
+
 bool target_vif_config_set2(const struct schema_Wifi_VIF_Config *vconf,
-			    const struct schema_Wifi_Radio_Config *rconf,
-			    const struct schema_Wifi_Credential_Config *cconfs,
```
Comment (blogic) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:985]: lots of code reformatting in this patch making it hard to see what has actually changed. could you please remove the reformatting part ?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 2]
```diff
@@ -980,119 +1006,172 @@ void vif_hs20_update(struct schema_Hotspot20_Config *hs2conf)
...
-		blobmsg_add_string(&b, "ssid", vconf->ssid);
+	vif_config_security_set(&b, vconf);
 
-	if (changed->ssid_broadcast) {
-		if (!strcmp(vconf->ssid_broadcast, "disabled"))
-			blobmsg_add_bool(&b, "hidden", 1);
-		else
-			blobmsg_add_bool(&b, "hidden", 0);
-	}
+	if (!strcmp(vconf->mode, "mesh")) {
```
Comment (blogic) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1064]: I am wondering if the code is more readable if we put this and the normal wifi setup each to their own function.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:985]: I think the whitespace change is to due to the mesh or ap decision resulting in the AP specific parameters adding 1 level of TAB to the whitespace.  The recommendation to separate the mesh and ap portions into two separate helper functions is probably the solution here.

**Review by RickSommerville — COMMENTED** [review]
Separate the mesh and AP into two separate functions called from target_vif_config_set2().  As John says, the readability will go way up and the changes will be more obvious.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:985]: Yeah. Already done with the changes. Currently resolving the merge conflicts with your latest commit. New patch should be pushed in a while.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 2 above)
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c]: Done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 1 above)
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:1001]: This one was badly indented. So had to change.

**Comment by blogic** [comment]
this appears to not apply ontop of latest pending.

**Comment by yashvardhan-netex** [comment]
> this appears to not apply ontop of latest pending.

Rebased.

**Comment by blogic** [comment]
merged into pending, Thanks !

**Pull request closed without merging**